core/unit_test/TestTeam.hpp

```diff
@@ -1642,15 +1642,16 @@ struct TestScratchAlignment {
           // and since scratch_ptr3 is then already aligned it difference
           // should match that
           if ((scratch_ptr3 - scratch_ptr2) != 32) flag() = 1;
+
           // check actually alignment of ptrs is as requested
-          if (((scratch_ptr1 % 4) != 0) || ((scratch_ptr2 % 8) != 0) ||
-              ((scratch_ptr3 % 4) != 0))
+          if ((int(scratch_ptr1 % 4) != 0) || (int(scratch_ptr2 % 8) != 0) ||
+              (int(scratch_ptr3 % 4) != 0))
```

Comment:
If they're all `intptr_t`, why do they need to be cast to `int` here?

Reply:
Is this just about the compiler mis-behaving?

Reply:
Yes.

Reply:
bloody hell, hate to do that to you but can you add a comment to that effect: i.e. we need to still cast to int here because SYCL compiler fails in mixed integer type comparisons with intptr_t ...

Also next week can we create a reproducer for Intel??

Reply:
what about a macro `KOKKOS_IMPL_SYCL_CAST_TO_INT`

Reply:
or some other ugly name that we only define when needed

Reply:
ping

Reply:
I added a comment, see https://github.com/kokkos/kokkos/pull/5767/files#diff-f1017e3e2c8e0566e5f1e0277a0150db93555c894cace36cb32aef5f75efb4ccR1647.